src/ui/toast.js

```diff
@@ -53,121 +39,96 @@ export const toastImportantStyles = {
+    setStyles(this.iframe_, topFriendlyIframePositionStyles);
+
+    /** @private @const {!Promise} */
+    this.ready_ = new Promise(resolve => {
+      this.iframe_.onload = resolve;
+    });
   }
 
+
   /**
-   * Gets the Iframe element.
+   * Returns the iframe element.
    * @return {!HTMLIFrameElement}
    */
   getElement() {
-    return this.iframe_.getElement();
+    return this.iframe_;
   }
 
   /**
    * Opens the notification toast.
    * @return {!Promise}
    */
   open() {
-    const iframe = this.iframe_;
-    if (iframe.isConnected()) {
-      throw new Error('Already opened');
-    }
-    this.doc_.body.appendChild(iframe.getElement());  // Fires onload.
+    this.doc_.body.appendChild(this.iframe_);  // Fires onload.
 
-    return iframe.whenReady().then(() => this.buildIframe_());
+    return this.buildToast_();
   }
 
   /**
-   * Closes the toast.
+   * Builds the content of the iframe. On load, animates the toast.
    */
-  close() {
-    this.doc_.body.removeChild(this.iframe_.getElement());
-  }
-
-  /**
-   * Builds the iframe with content and the styling after iframe is loaded.
-   * @private
-   * @return {!Toast}
-   */
-  buildIframe_() {
-    const iframe = this.iframe_;
-    const iframeDoc = iframe.getDocument();
-    const iframeBody = iframe.getBody();
-
-    // Inject Google fonts in <HEAD> section of the iframe.
-    injectFontsLink(iframeDoc, googleFontsUrl);
-    injectStyleSheet(iframeDoc, TOAST_CSS);
-
-    this.addItems_(iframeDoc, iframeBody);
-
-    return this;
+  buildToast_() {
+    return this.activityPorts_.openIframe(
+        this.iframe_, this.src_, this.args_).then(port => {
+          return port.whenReady();
+        }).then(() => {
+          resetStyles(this.iframe_, ['height']);
+          setImportantStyles(this.iframe_, {
+            'animation': 'swg-notify .3s ease-out normal backwards, '
+                  + 'swg-notify-hide .3s ease-out 7s normal forwards',
+          });
+        });
   }
 
   /**
-   * Adds label and detail button.
-   * @param {!Document} iframeDoc
-   * @param {?Element} iframeBody
-   * @private
+   * Closes the toast.
    */
-  addItems_(iframeDoc, iframeBody) {
-    const childElements = [];
-
-    const label = createElement(iframeDoc, 'div', {
-      'class': 'swg-label',
-    }, this.spec_.text);
-    childElements.push(label);
-
-    if (this.spec_.action && this.spec_.action.label) {
-      const linkButton = createElement(iframeDoc, 'button', {
-        'class': 'swg-detail',
-        'aria-label': 'Details',
-      }, this.spec_.action.label);
-      linkButton.addEventListener('click', this.spec_.action.handler);
-      childElements.push(linkButton);
-    }
-
-    // Create container element and add 'label' and/or 'linkButton' to it.
-    this.container_ = createElement(iframeDoc, 'div', {
-      'class': 'swg-toast-container',
-    }, childElements);
-
-    iframeBody.appendChild(this.container_);
+  close() {
```

Comment:
So, I'm still missing this answer: after animation is done animating out, the toast should somehow be removed from DOM. Where does it happen?

Reply:
Done. Added a callback after 8 seconds to remove the toast from dom. (Since toast is shown for 7 seconds).